### PR TITLE
Add initial implementation of task 2

### PR DESCRIPTION
**Pull request by azizumarov**

### CODE REVIEW

File: src/NetMPA.Catalog/NetMPA.Catalog.Bll.Tests/CatalogServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    {
+        [SetUp]
+        public void Setup()
+        {
+        }
+
+        [Test]
+        public void Test1()
+        {
+            Assert.Pass();
```
Comment (chulanov) [inline src/NetMPA.Catalog/NetMPA.Catalog.Bll.Tests/CatalogServiceTests.cs]: need real tests. At least for some methods

File: src/NetMPA.Catalog/NetMPA.Catalog.Bll/Interfaces/Repositories/IProductRepository.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NetMPA.Catalog.Bll.Interfaces.Repositories
+{
+    public interface IProductRepository
+    {
+        Task<Product> GetAsync(Guid id);
```
Comment (chulanov) [inline src/NetMPA.Catalog/NetMPA.Catalog.Bll/Interfaces/Repositories/IProductRepository.cs]: product id must be integer - Task 1 FR#4

File: src/NetMPA.Catalog/NetMPA.Catalog.Bll/Interfaces/Services/ICatalogService.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NetMPA.Catalog.Bll.Interfaces.Services
+{
+    public interface ICatalogService
```
Comment (chulanov) [inline src/NetMPA.Catalog/NetMPA.Catalog.Bll/Interfaces/Services/ICatalogService.cs]: Why do you need empty interface?

File: src/NetMPA.Catalog/NetMPA.Catalog.Bll/Services/CatalogService.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+using NetMPA.Catalog.Bll.Interfaces.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NetMPA.Catalog.Bll.Services
+{
+    public class CatalogService:ICatalogService
```
Comment (chulanov) [inline src/NetMPA.Catalog/NetMPA.Catalog.Bll/Services/CatalogService.cs]: need to implement business methods to complete the task

File: src/NetMPA.Catalog/NetMPA.Catalog.Dal.Tests/Repositories/CategoryRepositoryTests.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NetMPA.Catalog.Dal.Tests.Repositories
+{
+    internal class CategoryRepositoryTests
```
Comment (chulanov) [inline src/NetMPA.Catalog/NetMPA.Catalog.Dal.Tests/Repositories/CategoryRepositoryTests.cs]: need real tests

File: src/NetMPA.Catalog/NetMPA.Catalog.Dal/Repositories/CategoryRepository.cs [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        private readonly ICatalogContextFactory dbFactory;
+
+        public CategoryRepository(ICatalogContextFactory dbFactory)
+        {
+            this.dbFactory = dbFactory;
+        }
+
+        public Task AddAsync(Category category)
+        {
+            throw new NotImplementedException();
```
Comment (chulanov) [inline src/NetMPA.Catalog/NetMPA.Catalog.Dal/Repositories/CategoryRepository.cs]: need implementation to complete the task

**Review by chulanov — CHANGES_REQUESTED** [review]

File: src/NetMPA.Catalog/NetMPA.Catalog.Bll/Interfaces/Services/ICatalogService.cs [hunk 7]
```diff
@@ -8,6 +9,9 @@ namespace NetMPA.Catalog.Bll.Interfaces.Services
 {
     public interface ICatalogService
     {
+        Task<Product> GetProductById(int productId);
```
Comment (chulanov) [inline src/NetMPA.Catalog/NetMPA.Catalog.Bll/Interfaces/Services/ICatalogService.cs]: FR #3. The following operations are allowed for Category: get/list/add/update/delete.
FR #5. The following operations are allowed for Item: get/list/add/update/delete.

File: src/NetMPA.Catalog/NetMPA.Catalog.Bll/Models/Category.cs [hunk 8]
```diff
@@ -8,9 +8,9 @@ namespace NetMPA.Catalog.Bll.Models
 {
     public class Category
     {
-        public Guid Id { get; init; }
-        public string Name { get; init; }
-        public Uri Image { get; init; }
-        public Category Parent { get; init; }
+        public int Id { get; set; }
+        public string Name { get; set; }
```
Comment (chulanov) [inline src/NetMPA.Catalog/NetMPA.Catalog.Bll/Models/Category.cs:12]: FR #2.1 Name – required, plain text, max length = 50.

File: src/NetMPA.Catalog/NetMPA.Catalog.Bll/Models/Product.cs [hunk 9]
```diff
@@ -8,12 +8,12 @@ namespace NetMPA.Catalog.Bll.Models
...
     {
-        public Guid Id { get; init; }
-        public string Name { get; init; }
-        public string Description { get; init; }
-        public Uri Image { get; init; }
-        public Category Category { get; init; }
-        public decimal Price { get; init; }
-        public int Amount { get; init; }
+        public int Id { get; set; }
+        public string Name { get; set; } = string.Empty;
```
Comment (chulanov) [inline src/NetMPA.Catalog/NetMPA.Catalog.Bll/Models/Product.cs:12]: FR #4.1 Name – required, plain text, max length = 50.

File: src/NetMPA.Catalog/NetMPA.Catalog.Bll/Interfaces/Services/ICatalogService.cs (hunk 7 above)
Comment (azizumarov) [reply src/NetMPA.Catalog/NetMPA.Catalog.Bll/Interfaces/Services/ICatalogService.cs]: sure, will add

**Pull request merged**